components/Init/src/main.c

```diff
@@ -225,8 +225,11 @@ typedef struct memory_range {
 
 static memory_range_t guest_ram_regions[] = {
     /* Allocate all the standard low memory areas */
-    {0x500, 0x7c00 - 0x500},
-    {0x7e00, 0x80000 - 0x7e00},
+    /* {0x500, 0x7c00 - 0x500} and {0x7e00, 0x80000 - 0x7e00} are merged into one
```

Comment:
I might be lacking some historical context (our git history doesn't capture the introduction of these memory ranges). It seems `0x7C00`-`0x7DFF` was excluded to account for the MBR bootstrap code region. Though given we directly use Linux's 32-bit boot protocol (jumping directly into vmlinux entry point), I think the region should still be use-able memory in the end. 
As a minor nitpick, I think we could maybe extend the comment to name drop that we are now accounting for the MBR region, to give context as to why `{0x500, 0x7c00 - 0x500} and {0x7e00, 0x80000 - 0x7e00}` are merged into one.

Reply:
Yes, having some context here would be really helpful. Maybe even make a ASCII picture:
```
  --+--    +---------------------------+  0x00000     
    |      | 1280 bytes                |             
    |      +---------------------------+  0x00500     
    |      | 29184 byte                |            
    |      +---------------------------+  0x07700  (= 0x7c00 - 0x500)    
 512 KiB   | 1792 byte excluded        |
    |      | MBR is at 0x7C00-0x7E00   |
    |      +---------------------------+  0x07e00
    |      | 70400 byte                |             
    |      +---------------------------+  0x78200 (= 0x80000 - 0x7e00)
    |      | 32256 byte                |             
  --+--    +---------------------------+  0x80000
```